README.md

```diff
@@ -7,27 +7,33 @@
-that employs a serializer, and a compressor, handles all the other concerns including metadata and saving.
-Such that the serializer just needs to determine how to turn a julia object into a stream`Vector{UInt8}`,
-and the compressor just needs to determine how to turn one stream of `UInt8`s into a smaller one (and the reverse).
+Think of it less as a serialization format and more of a container for arbitrarily serialized objects and metadata.
+The modular structure of the JLSO container files allows users to balance their specific performance, flexibility and reliability requirements.
+
+A JLSO file contains two primary components:
 
+1. [BSON](https://bsonspec.org/spec.html) formatted metadata about the host operating system and environment
+2. An arbitrary collection of serialized objects
 
-At the top-level it is a BSON file,
-where it stores metadata about the system it was created on as well as a collection of objects (the actual data).
-Depending on configuration, those objects may themselves be stored as BSON sub-documents,
-or in the native Julia serialization format (default), under various levels of compression (`gzip` default).
-It is fast and efficient to load just single objects out of a larger file that contains many objects.
+While the metadata is specific to the JLSO file format, various serializers and compressors can be employed when saving objects.
+The serializer needs to determine how to turn a Julia object into a stream (`Vector{UInt8}`), while the compressor needs to choose how to convert one stream of `UInt8`s into a smaller one or the reverse.
+Depending on the configuration metadata, objects may be stored as [BSON sub-documents](https://bsonspec.org/spec.html) or [Julia serialized](https://docs.julialang.org/en/v1/stdlib/Serialization/) types, using various compression methods.
+JLSO currently defaults to [Julia serialization](https://docs.julialang.org/en/v1/stdlib/Serialization/) with [`gzip`](https://github.com/JuliaIO/CodecZlib.jl).
 
-The metadata includes the Julia version and the versions of all packages installed.
-It is always store in plain BSON without julia specific extensions.
-This means in the worst case you can install everything again and replicate your system.
-(Extreme worst case scenario, using a BSON reader from another programming language).
+Currently, JLSO files store the following metadata:
 
-Note: If the amount of data you have to store is very small, relative to the metadata about your environment, then JLSO is a pretty suboptimal format.
+- JLSO version number
+- Julia version number
+- Serialization format (e.g., "julia_serialize", "bson")
+- Compression method like "gzip"
+- A docker image URI
+- Project.toml
+- Manifest.toml (possibly compressed)
 
+This information makes it possible to reconstruct the original environment necessary to deserialize the objects if the internal serialization format has introduced breaking changes.
```

Comment:
I still don't love this sentence, but I want to make it clear that the metadata serves more as breadcrumbs for resolving serialization issues.

Reply:
Yes, I think it is not great.
Even if the serializer hasn't introduced breaking changes, if the structure of the object itself has changed that is also a problem, that need to be dealt with via loading the old object.

Maybe more like

```suggestion
In general, serializers are not able to handle deserialized objects if the type defintions of the stored objects have changed.
For example if a field has been added then there is no such field in the stored data so it can't be loaded using the new version of the package.
JLSO stores the package version, thus allowing you to load the old version of the package from before that change and so be able to recover the object.
Similar occurs if the serializer itself changes.
```

Even though it is longer